scipy/signal/_signaltools.py

```diff
@@ -93,6 +93,22 @@ def _inputs_swap_needed(mode, shape1, shape2, axes=None):
     return not ok1
 
 
+def _reject_objects(arr, name):
+    """Warn if arr.dtype is object or longdouble or float16.
+    """
+    dt = np.asarray(arr).dtype
+    if not (np.issubdtype(dt, np.integer)
+            or dt in [np.bool_, np.float16, np.float32, np.float64,
+                      np.complex64, np.complex128]
+    ):
+        msg = (
+            f"dtype={dt} is not supported by {name} and will raise an error in "
+            f"SciPy 1.17.0. Supported dtypes are: boolean, integer, np.float16,"
+            f"`np.float32`, `np.float64`, `np.complex64`, `np.complex128`."
+        )
```

Comment:
Let's make the backticks consistent here - either add them around `np.float16`, `dtype={dt}` and `{name}` and adjust the tests that check the message (my preference), or just remove them from the bottom line here (also okay).